infra/blueprint-test/pkg/golden/golden.go

```diff
@@ -37,9 +38,26 @@ const (
 )
 
 type GoldenFile struct {
-	dir      string
-	fileName string
-	t        testing.TB
+	dir        string
+	fileName   string
+	sanitizers []Sanitizer
+	t          testing.TB
+}
+
+type Sanitizer func(string) string
+
+// StringSanitizer replaces all occurrences of old string with new string
+func StringSanitizer(old, new string) Sanitizer {
+	return func(s string) string {
+		old := strings.ReplaceAll(s, old, new)
+		return old
+	}
+}
+
+// ProjectIDSanitizer replaces all occurrences of current gcloud project ID with PROJECT_ID string
```

Comment:
Should we include this by default?

Reply:
I am leaning towards no as we don't set gcloud config in container today. We can add this as a default later on if it becomes a common pattern.

Reply:
What's the behavior if it's not set? (Can we add a test for that?)

Assuming it doesn't cause any errors if unset, I suggest we use it be default.

Reply:
@morgante added as default and added a test. If its not set exit code is still 0 with an empty result. I added some additional logging for this case, so users know its skipping.